pkg/apis/networking/types.go

```diff
@@ -90,6 +90,25 @@ type NetworkPolicyPort struct {
 	Port *intstr.IntOrString
 }
 
+// CIDR describes a Network CIDR to select
+type CIDR string
+
+// Except is a slice of CIDRs to not include within an IpBlockRule
+type Except []string
+
+// IpBlockRule describes a particular CIDR (Ex. "192.168.1.1/24") that is allowed to the pods
+// matched by a NetworkPolicySpec's podSelector. The except entry describes CIDRs that should
+// not be included within this rule.
+type IpBlockRule struct {
+	// CIDR is a string representing a Classless inter-domain route
+	// Valid examples are "192.168.1.1/24"
+	CIDR string
+	// Except is a slice of CIDRs that should not be included within an IPBlock rule
```

Comment:
You're right. I have no idea why I came up with rule and it becomes terribly confusing. I switched the the networking.IPBlock to represent the type for a IPBlockSelector. 

Should we continue to use something like ipBlock in the json or ipBlockSelector?